src/Domain/FunctionsOfTime/IntegratedFunctionOfTime.cpp

```diff
@@ -0,0 +1,143 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/FunctionsOfTime/IntegratedFunctionOfTime.hpp"
+
+#include <algorithm>
+#include <iterator>
+#include <memory>
+#include <ostream>
+
+#include "DataStructures/DataVector.hpp"
+#include "Domain/FunctionsOfTime/ThreadsafeList.tpp"
+#include "Utilities/ErrorHandling/Error.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain::FunctionsOfTime {
+
+IntegratedFunctionOfTime::IntegratedFunctionOfTime() = default;
+IntegratedFunctionOfTime::IntegratedFunctionOfTime(IntegratedFunctionOfTime&&) =
+    default;
+IntegratedFunctionOfTime::IntegratedFunctionOfTime(
+    const IntegratedFunctionOfTime&) = default;
+auto IntegratedFunctionOfTime::operator=(IntegratedFunctionOfTime&&)
+    -> IntegratedFunctionOfTime& = default;
+auto IntegratedFunctionOfTime::operator=(const IntegratedFunctionOfTime&)
+    -> IntegratedFunctionOfTime& = default;
+IntegratedFunctionOfTime::~IntegratedFunctionOfTime() = default;
+
+IntegratedFunctionOfTime::IntegratedFunctionOfTime(
+    const double t, std::array<double, 2> initial_func_and_derivs,
+    const double expiration_time, bool rotation)
+    : deriv_info_at_update_times_(t), rotation_(rotation) {
+  deriv_info_at_update_times_.insert(t, std::move(initial_func_and_derivs),
+                                     expiration_time);
+}
+
+std::unique_ptr<FunctionOfTime> IntegratedFunctionOfTime::get_clone() const {
+  return std::make_unique<IntegratedFunctionOfTime>(*this);
+}
+
+template <size_t MaxDerivReturned>
+std::array<DataVector, MaxDerivReturned + 1>
+IntegratedFunctionOfTime::func_and_derivs(const double t) const {
+  static_assert(MaxDerivReturned <= 1, "can only return 2 derivatives");
+
+  auto deriv_info_at_t = deriv_info_at_update_times_(t);
```

Comment:
ERROR if `deriv_info_at_t.expiration != t and t != deriv_info_at_times_.initial_time()`.

Reply:
do you mean greater than expiration or lower than initial? `ThreadSafeList` throws an error but I can add another one here.

Reply:
No, I mean what I wrote.  This class doesn't work correctly with dense output, so you should check that it is not being used for such.

Reply:
Right, but I will set the update times in between the substep times. This way I dont have to worry about which interval will be selected at the boundary due to roundoff.

Reply:
OK.  This is fine as-is, then.